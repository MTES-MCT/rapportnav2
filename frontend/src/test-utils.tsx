import React, { ReactElement } from 'react'
import { render, renderHook, RenderOptions } from '@testing-library/react'
import UIThemeWrapper from './ui/ui-theme-wrapper'
import { BrowserRouter } from 'react-router-dom'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <UIThemeWrapper>
      <BrowserRouter>{children}</BrowserRouter>
    </UIThemeWrapper>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options })

const customRenderHook = (hook: any, options?: Omit<RenderOptions, 'wrapper'>) =>
  renderHook(() => hook(), { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
export { customRenderHook as renderHook }
