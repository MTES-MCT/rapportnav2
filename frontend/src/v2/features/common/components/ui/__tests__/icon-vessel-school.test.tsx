import { render } from '../../../../../../test-utils'
import IconVesselSchool from '../icon-vessel-school'

describe('IconVesselSchool', () => {
  it('should match the snapshot', () => {
    const wrapper = render(<IconVesselSchool />)
    expect(wrapper).toMatchSnapshot()
  })
})
