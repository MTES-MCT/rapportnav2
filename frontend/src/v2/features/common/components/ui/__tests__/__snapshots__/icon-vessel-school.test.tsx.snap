// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IconVesselSchool > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        alt="Bateaux école"
        src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20width='55'%20height='28'%20viewBox='0%200%2055%2028'%3e%3cdefs%3e%3cclipPath%20id='clip-path'%3e%3crect%20id='Rectangle_6789'%20width='59.942'%20height='28.89'%20transform='translate(-5%20-5)'%20fill='%233b4559'/%3e%3c/clipPath%3e%3cclipPath%20id='clip-Plan_de_travail_2'%3e%3crect%20width='55'%20height='28'/%3e%3c/clipPath%3e%3c/defs%3e%3cg%20id='Plan_de_travail_2'%20clip-path='url(%23clip-Plan_de_travail_2)'%3e%3crect%20width='55'%20height='28'%20fill='transparent'/%3e%3cg%20id='Groupe_5543'%20transform='translate(1%203.918)'%3e%3cg%20id='Groupe_5526'%20transform='translate(0%200)'%20clip-path='url(%23clip-path)'%3e%3cpath%20id='Tracé_1807'%20d='M30.875,2.028A4.808,4.808,0,0,0,28.007,0H8.242V3.436l2.3,1.282L8.264,12.574l-8.154.512L0,13.093l.07.3,29.41-1.842,6.309-3.338.1-.053ZM27.728,1.14l1.156,1.245L9.391,2.267V1.14Zm-4.72,3.578-1.378,5.543-7.91.466,1.808-6.008ZM33.8,7.74l-4.13,2.154-6.5.362,1.608-5.539H31.13Z'%20transform='translate(0%20-1)'%20fill='%233b4559'/%3e%3cpath%20id='Tracé_1808'%20d='M37.361,9.1,29.628,13.19.073,15.041,2,23.109h34.9c.038,0,.2.008.452.008,1.56,0,6.694-.222,9.5-3.024,3.24-3.239,4.974-11.4,5.046-11.747l.053-.258ZM6.8,16.907H4.755v1.483H6.147a.559.559,0,1,1,0,1.117H4.755V20.99H6.8a.559.559,0,0,1,0,1.119H4.2a.554.554,0,0,1-.559-.559v-5.2a.554.554,0,0,1,.559-.56H6.8a.559.559,0,0,1,0,1.118m4.987,1.3a.554.554,0,0,1-.559-.559.742.742,0,0,0-1.483,0v2.6a.742.742,0,0,0,1.483,0,.559.559,0,1,1,1.118,0,1.86,1.86,0,0,1-3.719,0v-2.6a1.86,1.86,0,0,1,3.719,0,.553.553,0,0,1-.559.559M13.83,20.25v-2.6a1.86,1.86,0,0,1,3.719,0v2.6a1.86,1.86,0,0,1-3.719,0m7.931.74a.559.559,0,0,1,0,1.119H19.594a.554.554,0,0,1-.56-.559v-5.2a.559.559,0,1,1,1.118,0V20.99Zm5.2-4.083H24.923v1.483h1.392a.559.559,0,1,1,0,1.117H24.923V20.99h2.042a.559.559,0,0,1,0,1.119h-2.6a.554.554,0,0,1-.56-.559v-5.2a.554.554,0,0,1,.56-.56h2.6a.553.553,0,0,1,.559.56.559.559,0,0,1-.559.558'%20transform='translate(-0.011%20-1.228)'%20fill='%233b4559'/%3e%3cpath%20id='Tracé_1809'%20d='M18.344,22.781A.783.783,0,0,0,19.125,22V19.031a.782.782,0,1,0-1.563,0V22a.783.783,0,0,0,.782.782'%20transform='translate(-2.666%20-2.77)'%20fill='%233b4559'/%3e%3c/g%3e%3c/g%3e%3c/g%3e%3c/svg%3e"
      />
    </div>
    <div
      class="rs-toast-provider"
    >
      <div
        class="rs-toast-container rs-toast-container-top-center"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-center"
      />
      <div
        class="rs-toast-container rs-toast-container-top-start"
      />
      <div
        class="rs-toast-container rs-toast-container-top-end"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-start"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-end"
      />
    </div>
  </body>,
  "container": <div>
    <img
      alt="Bateaux école"
      src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20width='55'%20height='28'%20viewBox='0%200%2055%2028'%3e%3cdefs%3e%3cclipPath%20id='clip-path'%3e%3crect%20id='Rectangle_6789'%20width='59.942'%20height='28.89'%20transform='translate(-5%20-5)'%20fill='%233b4559'/%3e%3c/clipPath%3e%3cclipPath%20id='clip-Plan_de_travail_2'%3e%3crect%20width='55'%20height='28'/%3e%3c/clipPath%3e%3c/defs%3e%3cg%20id='Plan_de_travail_2'%20clip-path='url(%23clip-Plan_de_travail_2)'%3e%3crect%20width='55'%20height='28'%20fill='transparent'/%3e%3cg%20id='Groupe_5543'%20transform='translate(1%203.918)'%3e%3cg%20id='Groupe_5526'%20transform='translate(0%200)'%20clip-path='url(%23clip-path)'%3e%3cpath%20id='Tracé_1807'%20d='M30.875,2.028A4.808,4.808,0,0,0,28.007,0H8.242V3.436l2.3,1.282L8.264,12.574l-8.154.512L0,13.093l.07.3,29.41-1.842,6.309-3.338.1-.053ZM27.728,1.14l1.156,1.245L9.391,2.267V1.14Zm-4.72,3.578-1.378,5.543-7.91.466,1.808-6.008ZM33.8,7.74l-4.13,2.154-6.5.362,1.608-5.539H31.13Z'%20transform='translate(0%20-1)'%20fill='%233b4559'/%3e%3cpath%20id='Tracé_1808'%20d='M37.361,9.1,29.628,13.19.073,15.041,2,23.109h34.9c.038,0,.2.008.452.008,1.56,0,6.694-.222,9.5-3.024,3.24-3.239,4.974-11.4,5.046-11.747l.053-.258ZM6.8,16.907H4.755v1.483H6.147a.559.559,0,1,1,0,1.117H4.755V20.99H6.8a.559.559,0,0,1,0,1.119H4.2a.554.554,0,0,1-.559-.559v-5.2a.554.554,0,0,1,.559-.56H6.8a.559.559,0,0,1,0,1.118m4.987,1.3a.554.554,0,0,1-.559-.559.742.742,0,0,0-1.483,0v2.6a.742.742,0,0,0,1.483,0,.559.559,0,1,1,1.118,0,1.86,1.86,0,0,1-3.719,0v-2.6a1.86,1.86,0,0,1,3.719,0,.553.553,0,0,1-.559.559M13.83,20.25v-2.6a1.86,1.86,0,0,1,3.719,0v2.6a1.86,1.86,0,0,1-3.719,0m7.931.74a.559.559,0,0,1,0,1.119H19.594a.554.554,0,0,1-.56-.559v-5.2a.559.559,0,1,1,1.118,0V20.99Zm5.2-4.083H24.923v1.483h1.392a.559.559,0,1,1,0,1.117H24.923V20.99h2.042a.559.559,0,0,1,0,1.119h-2.6a.554.554,0,0,1-.56-.559v-5.2a.554.554,0,0,1,.56-.56h2.6a.553.553,0,0,1,.559.56.559.559,0,0,1-.559.558'%20transform='translate(-0.011%20-1.228)'%20fill='%233b4559'/%3e%3cpath%20id='Tracé_1809'%20d='M18.344,22.781A.783.783,0,0,0,19.125,22V19.031a.782.782,0,1,0-1.563,0V22a.783.783,0,0,0,.782.782'%20transform='translate(-2.666%20-2.77)'%20fill='%233b4559'/%3e%3c/g%3e%3c/g%3e%3c/g%3e%3c/svg%3e"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
